### PR TITLE
Supportconfig Testcase

### PR DESCRIPTION
**Pull request by basheerk**
The supportconfig script gathers system troubleshooting information on
SUSE Linux Enterprise Server systems.It captures the current system
environment and generates a tar-archive

Signed-off-by: Basheer K<basheer@linux.vnet.ibm.com>
Tested-by: Basheer K<basheer@linux.vnet.ibm.com>

### CODE REVIEW

File: toolchain/supportconfig.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    """
+
+    def setUp(self):
+        """
+        Install all the dependency packages required for Test.
+        """
+        sm = SoftwareManager()
+        dist = distro.detect()
+
+        if "SuSE" not in dist.name:
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: please use distro.detect().name 

` if "SuSE" not in distro.detect().name:`

as for extra variable python garbage collector need to invoke

File: toolchain/supportconfig.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            self.skip("supportconfig is supported on the SuSE Distro only.")
+
+        package = "supportutils"
+        if not sm.check_installed(package) and not sm.install(package):
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: please here as well 

`if not sm.check_installed("supportutils") and not sm.install("supportutils"):`

File: toolchain/supportconfig.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                                  (option, self.workdir),
+                                  sudo=True,
+                                  ignore_status=True)
+                if self.workdir not in ret.stdout or ret.exit_status:
+                    self.fail(
+                        "Failed to change the supportconfig output" +
+                        " files directory")
+            else:
+                self.log.info("verifying %s option of supportconfig" % option)
+                ret = process.run("supportconfig -%s" % option,
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: process.system() can be used as it return exit code

File: toolchain/supportconfig.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    def test_enable_disable_plugins(self):
+        """
+        1.create the /usr/lib/supportconfig/plugins
+        2.copy your script file there
+        3.output will be in the plugin-plugin_name.txt file
+        """
+        plugin_dir = "/usr/lib/supportconfig/plugins"
+        plugin_dir_exists = 1
+        if not os.path.exists(plugin_dir):
+            plugin_dir_exists = 0
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: i think  `plugin_dir_exists `variable is not required what you think ?

File: toolchain/supportconfig.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            os.mkdir(plugin_dir)
+        # copy the plugin file
+        plugin_name = '/usr/bin/pstree'
+        process.system("cp %s %s" % (plugin_name, plugin_dir))
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: please use shutil python lib for copy files 

code can be 
`shutil.copy('/usr/bin/pstree', '/usr/lib/supportconfig/plugins')`

File: toolchain/supportconfig.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        ret = process.run("supportconfig",
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: please use `process.system()` as it return exit code

File: toolchain/supportconfig.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                          sudo=True)
+        logfile = re.search(r"Log file tar ball: (\S+)\n", ret.stdout).group(1)
+        res = process.system("tar -tvf %s | grep 'plugin-pstree.txt'"
+                             % logfile,
+                             ignore_status=True,
+                             shell=True)
+        if ret.exit_status or res:
+            self.fail("support failed to execute plugin")
+        # disable a plugin
+        ret = process.run("supportconfig -p",
```
Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: please here also `process.system()`

File: toolchain/supportconfig.py (hunk 1 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: accepted..will modify as suggested

File: toolchain/supportconfig.py (hunk 5 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: accepted..will modify as suggested

File: toolchain/supportconfig.py (hunk 6 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: can't replace process.run() with process.system() , as i am parsing the output from the process.run() to get logile.

File: toolchain/supportconfig.py (hunk 7 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: same as above

File: toolchain/supportconfig.py (hunk 4 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: @PraveenPenguin I am using plugin_dir_exists variable to cleanup the plugin_dir,if the directory doesn't exist by default.

Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: @basheerk still i am confused as clean up section you written here 
https://github.com/basheerk/avocado-misc-tests-1/blob/0b2f08d76f97817b91d1dc03ff7ee65aa5a5f2c1/toolchain/supportconfig.py#L137

this code 
```
        if not os.path.exists(plugin_dir):
            plugin_dir_exists = 0
```
executed is if directory not present

File: toolchain/supportconfig.py (hunk 6 above)
Comment (basheerk) [inline toolchain/supportconfig.py]: Below is the snippet of the code,for getting the logfile i am parsing the output from process.run() 
        ret = process.run("supportconfig",
                          sudo=True)
        logfile = re.search(r"Log file tar ball: (\S+)\n", ret.stdout).group(1)

Comment (PraveenPenguin) [inline toolchain/supportconfig.py]: oh sure

**Comment by basheerk** [comment]
@PraveenPenguin updated the changes

**Comment by PraveenPenguin** [comment]
@basheerk thanks can you please attach job log

**Comment by basheerk** [comment]
attaching job log file
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/593828/job.txt)

**Pull request merged**